lingua_franca/lang/parse_en.py

```diff
@@ -1383,29 +1421,53 @@ def date_found():
             temp = datetime.strptime(datestr, "%B %d")
         except ValueError:
             # Try again, allowing the year
-            temp = datetime.strptime(datestr, "%B %d %Y")
-        extractedDate = extractedDate.replace(hour=0, minute=0, second=0)
-        if not hasYear:
-            temp = temp.replace(year=extractedDate.year,
-                                tzinfo=extractedDate.tzinfo)
-            if extractedDate < temp:
-                extractedDate = extractedDate.replace(
-                    year=int(currentYear),
-                    month=int(temp.strftime("%m")),
-                    day=int(temp.strftime("%d")),
-                    tzinfo=extractedDate.tzinfo)
+            try:
+                temp = datetime.strptime(datestr, "%B %d %Y")
+            except:
+                temp = None
+        if temp is None:
+            # check for month
+            for m in months + monthsShort:
+                if m in datestr:
+                    int_month = month_to_int_en(m)
+                    year = extractedDate.year
+                    if int_month < extractedDate.month:
```

Comment:
* "March 15th" returns Mar 15 of next year (OK)
* "Next March 15th" returns Mar 1 of THIS year, (which, in addition to being the wrong date, already happened)
* "This March 15th" also returns Mar 1 of this year
* "Last March 15th" returns March 1 of last year

Also, how would one (intentionally) ask for the March 15th that just happened?